test/groovy/org/ods/PipelineMock.groovy

```diff
@@ -0,0 +1,28 @@
+package org.ods
+
+class PipelineMock {
```

Comment:
Let's do it in a simpler way. You can double-check that the steps parameter you are passing implements IPipelineSteps. Add a withCredentials Method to IPipelineSteps and its implementations. Then just create a Stub(IPipelineSteps) for the test methods.